javascript-source/core/permissions.js

```diff
@@ -208,61 +220,89 @@
      * @returns {boolean}
      */
     function isAdmin(username) {
-        return getUserGroupId(username.toLowerCase()) <= 1 || isOwner(username);
+        return queryDBPermission(username.toLowerCase()) <= PERMISSION.Admin || isOwner(username);
+    }
+
+    /**
+     * @function isModNoTags
+     * @export $
+     * @param {string} username
+     * @returns {boolean}
+     */
+    function isModNoTags(username) {
+        return isModeratorCache(username.toLowerCase()) || queryDBPermission(username.toLowerCase()) <= PERMISSION.Mod || isOwner(username);
+    }
+
+    function checkTags(tags) {
+        return tags !== null && tags != '{}' && tags != '-1' && tags !== undefined;
     }
 
     /**
      * @function isMod
      * @export $
      * @param {string} username
+     * @param {Object} tags
      * @returns {boolean}
      */
-    function isMod(username) {
-        return isModeratorCache(username.toLowerCase()) || getUserGroupId(username.toLowerCase()) <= 2 || isOwner(username);
+    function isMod(username, tags) {
+        if (checkTags(tags) && tags.get('user-type').length() > 0) { // Broadcaster should be included here.
+            return true;
+        }
+
+        $.consoleDebug('Used isMod without tags::' + tags);
+        return isModNoTags(username);
     }
 
     /**
-     * @function isModv3
+     * @function isMod
+     * @deprecated
+     *
      * @export $
      * @param {string} username
      * @param {Object} tags
      * @returns {boolean}
      */
     function isModv3(username, tags) {
-        if (tags != null && tags != '{}') {
-            if (tags.get('user-type').length() > 0) { // Broadcaster should be included here.
-                return true;
-            }
-        }
-        $.consoleDebug('Used isModv3 without tags::' + tags);
-        return isMod(username);
+        return isMod(username, tags);
+    }
+
+    /**
+     * @function isSubNoTags
+     * @export $
+     * @param {string} username
+     * @returns {boolean}
+     */
+    function isSubNoTags(username) {
+        return subUsers.contains(java.util.Objects.toString(username.toLowerCase())) || queryDBPermission(username.toLowerCase()) === PERMISSION.Sub;
     }
 
     /**
      * @function isSub
      * @export $
      * @param {string} username
+     * @param {Object} tags
      * @returns {boolean}
      */
-    function isSub(username) {
-        return subUsers.contains(java.util.Objects.toString(username.toLowerCase()));
+    function isSub(username, tags) {
+        if (checkTags(tags) && tags.containsKey('subscriber')) {
+            return tags.get('subscriber').equals('1');
+        }
+
+        $.consoleDebug('Used isSub without tags::' + tags);
+        return isSubNoTags(username);
     }
 
     /**
      * @function isSubv3
+     * @deprecated
+     *
      * @export $
      * @param {string} username
      * @param {Object} tags
      * @returns {boolean}
      */
     function isSubv3(username, tags) {
```

Comment:
Function is a pass-through to `isSub` with same signature, can be removed and re-mapped